IPython/core/magics/execution.py

```diff
@@ -1261,13 +1261,23 @@ def time(self,line='', cell=None, local_ns=None):
         # Minimum time above which compilation time will be reported
         tc_min = 0.1
 
+        expr_val=None
         if len(expr_ast.body)==1 and isinstance(expr_ast.body[0], ast.Expr):
             mode = 'eval'
             source = '<timed eval>'
             expr_ast = ast.Expression(expr_ast.body[0].value)
         else:
             mode = 'exec'
             source = '<timed exec>'
+            # multi-line %%time case
+            if len(expr_ast.body) > 1 :
+                expr_val=expr_ast.body[-1]
+                code_val = self.shell.compile(ast.Expression(expr_val.value)
+                                                , '<timed eval>'
+                                                , 'eval')
+                expr_ast=expr_ast.body[:-1]
+                expr_ast = ast.Module(expr_ast)
```

Comment:
That apparently is not compatible python 3.8+; I'd love to not break on a new Python.

Reply:
You need to do something similar to https://github.com/ipython/ipython/blob/master/IPython/core/interactiveshell.py#L110-L116

Reply:
I've done it.